### PR TITLE
Fix columns duplication on MongoDB Query Runner #6640

### LINKED ISSUE #6640: MongoDB Query Runner displays duplicated columns
### Issue Summary

MongoDB Query Runner displays duplicated columns on querying without `fields`.

### Steps to Reproduce

1. Insert multiple records
```
test> db.hoge.find()
[
  { _id: ObjectId('6569ecd1214c2278faf58ca3'), name: 'test' },
  { _id: ObjectId('6569ee53d53db7930aaa0cc0'), name: 'test2', age: 10 }
]
```

2. Send a query without `fields`
```
{"collection": "hoge"}
```

3. The columns are "_id", "name", "_id", "name", "age". The expected columns are  "_id", "name", "age".
image

### Technical details:

* Redash Version: master branch
* Browser/OS: Chrome / Docker
* How did you install Redash:

### PR DESCRIPTION
**Pull request by masayuki038**
## What type of PR is this? 

- [x] Bug Fix

## Description
Bug fix for https://github.com/getredash/redash/issues/6640.
And added some tests for MongoDB Query Runner.

## How is this tested?

- [x] Unit tests (pytest, jest)
- [ ] E2E Tests (Cypress)
- [x] Manually
- [ ] N/A

## Related Tickets & Documents
Closes #6640

### without `fields`
image

### with `fields`
image

### CODE REVIEW

**Review by konnectr — APPROVED** [review]

**Comment by guidopetri** [comment]
@masayuki038 thanks for this fix. It looks like two of the tests aren't passing for it? Could you take a look and investigate why?

**Review by guidopetri — CHANGES_REQUESTED** [review]
@masayuki038 thanks for this fix. It looks like two of the tests aren't passing for it? Could you take a look and investigate why?

**Comment by masayuki038** [comment]
@guidopetri @konnectr Thanks for your review.

> It looks like two of the tests aren't passing for it? Could you take a look and investigate why?

Sure. I missed something. I'll check it later. Please give me a little time.

File: redash/query_runner/mongodb.py [hunk 1]
```diff
@@ -113,13 +113,14 @@ def parse_results(results):
...
         parsed_row = _parse_dict(row)
         for column_name, value in parsed_row.items():
-            columns.append(
-                {
-                    "name": column_name,
-                    "friendly_name": column_name,
-                    "type": TYPES_MAP.get(type(value), TYPE_STRING),
-                }
-            )
+            if _get_column_by_name(columns, column_name) is None:
```
Comment (clearnote01) [inline redash/query_runner/mongodb.py:116]: To make this more efficient, can't we just get the first row, and then generate the columns array from that outside for this for loop? 

In this case every cell in the table requires this comparison which is in-efficient and wether we check row1 or rowN the columns will be same.

Comment (masayuki038) [reply redash/query_runner/mongodb.py:116]: @clearnote01 I understand your comment. I also think that if it is normal RDB data, I should do that.

I thought that MongoDB might have a different column for each row, so I implemented it like this. Does such a case not exist?

Comment (clearnote01) [reply redash/query_runner/mongodb.py:116]: @masayuki038 You are right, I don't think it's a problem when projection is specifically configured via $project, but there could be other cases that would break it. 

Maybe it's safer to keep it as it is, and have a note that this could be optimized in the future if someone can rule out the edge cases.

**Review by clearnote01 — APPROVED** [review]

**Comment by masayuki038** [comment]
e2e test still failed. I don't know main branch status... 
I will re-run it after https://github.com/getredash/redash/pull/6748 is fixed.

**Comment by masayuki038** [comment]
e2e test passed by merging main branch.
@guidopetri Could you check this if you have some time? Thank you always.

**Comment by jagal17** [comment]
@clearnote01 any news about this merge? Please we need it!

**Comment by masayuki038** [comment]
Sorry, something is wrong. I will check it. Please wait.

**Comment by justinclift** [comment]
@wlach Do you have time/interest to look over this one?  :smile:

**Comment by justinclift** [comment]
@eradman Do you have time to look over this one?  It seems a bit more urgent than some of the other PRs around, as people are emailing me directly about this one to ask. :wink:

**Comment by eradman** [comment]
> @eradman Do you have time to look over this one? It seems a bit more urgent than some of the other PRs around, as people are emailing me directly about this one to ask. 😉

I don't have enough knowledge about MongoDB to provide useful input on this. Perhaps merge it and ask for feedback?

**Comment by justinclift** [comment]
Good idea, lets do that. :smile:

**Review by justinclift — APPROVED** [review]
As @eradman said, we'll merge this.

@jagal17 Are you ok to verify a build (with this merged) works for you?

**Comment by justinclift** [comment]
@masayuki038 Thanks for taking the time to get this over the line. :smile:

**Comment by masayuki038** [comment]
@justinclift @eradman Thanks for your reviews! I checked it on local, but let me know if someone reported some issues around this. I am checking github every day 😄 Thank you.

**Comment by jagal17** [comment]
@justinclift @masayuki038 it is working very well now! thank you very much guys!
